main.go

```diff
@@ -72,60 +75,44 @@ func main() {
 			os.Exit(1)
 		}
 	}
+
 	ip, err := getHostIp()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Warning: failed to determine host's ip: %s\n", err)
-		ip = ""
 	}
+
+	deploymentType := getDeploymentType()
+
 	// Retrieve zone from the metadata server only if not specified in args.
-	zone := *localityZone
-	if zone == "" {
-		zone, err = getZone()
+	if *localityZone == "" {
+		*localityZone, err = getZone()
```

Comment:
I'm not wild about modifying the flags; that's modifying global state. Keep this as it was? Ditto for gkeClusterName and elsewhere.